client-lite/src/exe/docs.cpp

```diff
@@ -16,57 +16,95 @@ namespace msdoutil = microsoft::deliveryoptimization::util::details;
 
 #include "config_manager.h"
 #include "download_manager.h"
+#include "do_noncopyable.h"
 #include "proc_launch_helper.h"
 #include "rest_http_controller.h"
 #include "rest_port_advertiser.h"
 #include "trace_sink.h"
 
 using namespace std::chrono_literals; // NOLINT(build/namespaces) how else should we use chrono literals?
 
-class ProcessController
+class ProcessController : DONonCopyable
 {
 public:
     ProcessController()
     {
-        // Do init work like registering signal control handler
-        signal(SIGINT, _SignalHandler);
-        signal(SIGTERM, _SignalHandler);
     }
 
-    void WaitForShutdown(const std::function<bool()>& fnIsIdle)
+    void SetSignalHandler(const std::function<void()>& fnHandler, int signalNumber)
     {
+        if (signalNumber == SIGINT)
+        {
+            _sigintHandler = std::move(fnHandler);
+        }
+        else if (signalNumber == SIGTERM)
+        {
+            _sigtermHandler = std::move(fnHandler);
+        }
+        else if (signalNumber == SIGHUP)
+        {
+            _sighupHandler = std::move(fnHandler);
+        }
```

Comment:
>} [](start = 8, length = 1)

Depending on this will change based on my refactoring comment, add an else case here with DO_ASSERT(false).

Reply:
Think I'll revert the changes and just make the constructor accept the callback for refreshing configs

---
In reply to: [595419554](https://github.com/microsoft/do-client/pull/11#discussion_r595419554) [](ancestors = 595419554)